### PR TITLE
feat: add log4j2 configuration

### PR DESCRIPTION
**Pull request by OrezzerO**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance

- Related issues

### New feature or improvement
- Describe the details and related test reports.
 Add log4j configuration. The logs will be at `${user.dir}/logs/java-runner-common-default.log` and `${user.dir}/logs/java-runner-common-error.log`.  In apisix env, `${user.dir}` generally equals `/usr/local/apisix/`.

- Source branch

- Related commits and pull requests

- Target branch

Related issue: #22

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -88,6 +88,11 @@
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
+            <dependency>
+                <groupId>com.lmax</groupId>
+                <artifactId>disruptor</artifactId>
+                <version>3.4.4</version>
+            </dependency>
```
Comment (tzssangglass) [inline pom.xml:95]: Is this a complete framework or just a logging component?
What is the link to its repository?

Comment (OrezzerO) [reply pom.xml:95]: [disruptor](https://lmax-exchange.github.io/disruptor/) is a "High Performance Inter-Thread Messaging Library" . Log4j2 uses this lib to implement async logging.

Comment (tzssangglass) [reply pom.xml:95]: With this lib added, will the java runner logs be output in APISIX?

Comment (OrezzerO) [reply pom.xml:95]: It is `log4j2.xml`  decides whether runner's logs be output in  APISIX. Now runners log will not be send to APISIX. If users want to send logs to apisix, they need add following code in `log4j2.xml`:


Add a appender:
```xml
<Console name="CONSOLE" target="SYSTEM_OUT" follow="true">
    <PatternLayout charset="${FILE_ENCODING}">
        <pattern>%d %-5p %-32t - %m%n %throwable</pattern>
    </PatternLayout>
</Console>
```

Add appender above to loggers:
```xml
<appender-ref ref="CONSOLE"/>
```

Comment (tzssangglass) [reply pom.xml:95]: do we need to set the system property `log4j2.contextSelector` to `org.apache.logging.log4j.core.async.AsyncLoggerContextSelector` ref: https://logging.apache.org/log4j/2.x/manual/async.html

**Comment by tzssangglass** [comment]
we need to add log path to https://github.com/apache/apisix-java-plugin-runner/blob/main/.gitignore#L25

File: pom.xml (hunk 1 above)
Comment (OrezzerO) [reply pom.xml:95]: Already set `<asyncRoot>`, so there is no need to set system property "log4j2.contextSelector" to any value.

File: runner-starter/src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        </RollingFile>
+    </Appenders>
+
+    <Loggers>
+        <AsyncRoot level="${LOG_LEVEL}">
+            <appender-ref ref="ROOT-APPENDER"/>
+            <appender-ref ref="ERROR-APPENDER"/>
+        </AsyncRoot>
+    </Loggers>
+</configuration>
```
[the remark this answers is not in the document]
Comment (OrezzerO) [reply runner-starter/src/main/resources/log4j2.xml]: ok

**Review by tzssangglass — APPROVED** [review]

**Comment by tzssangglass** [comment]
@OrezzerO merged, many thanks.

**Pull request merged**